File: src/main/java/com/docs/viewer/global/common/utils/FileTypeUtil.java
package com.docs.viewer.global.common.utils;

import org.springframework.http.MediaType;
import org.springframework.stereotype.Component;

import java.util.stream.Stream;

@Component
public class FileTypeUtil {

    private static final MediaType PDF = MediaType.APPLICATION_PDF;
    private static final MediaType DOCX = MediaType.valueOf("application/vnd.openxmlformats-officedocument.wordprocessingml.document");
    private static final MediaType PPTX = MediaType.valueOf("application/vnd.openxmlformats-officedocument.presentationml.presentation");
    private static final MediaType XLSX = MediaType.valueOf("application/vnd.openxmlformats-officedocument.spreadsheetml.sheet");
    private static final MediaType JPEG = MediaType.IMAGE_JPEG;
    private static final MediaType PNG = MediaType.IMAGE_PNG;

    public static boolean isNotAllowType(String mimeType) {
        return !isAllowType(mimeType);
    }

    public static boolean isAllowType(String mimeType) {
        MediaType type = MediaType.valueOf(mimeType);
        return Stream.of(
                PDF, DOCX, PPTX, XLSX, JPEG, PNG
        ).anyMatch(allowedType -> allowedType.includes(type));
    }

    public static boolean isPdf(String contentType) {
        return PDF.includes(MediaType.valueOf(contentType));
    }

    private static boolean isDocx(String contentType) {
        return DOCX.includes(MediaType.valueOf(contentType));
    }

    private static boolean isPptx(String contentType) {
        return PPTX.includes(MediaType.valueOf(contentType));
    }

    private static boolean isXlsx(String contentType) {
        return XLSX.includes(MediaType.valueOf(contentType));
    }
}

File: dotnet-docs-viewer/Common/Utils/FileTypeUtil.cs
using Microsoft.Net.Http.Headers;

namespace Common.Utils
{
    public static class FileTypeUtil
    {
        private static readonly MediaTypeHeaderValue Pdf = MediaTypeHeaderValue.Parse("application/pdf");
        private static readonly MediaTypeHeaderValue Docx = MediaTypeHeaderValue.Parse("application/vnd.openxmlformats-officedocument.wordprocessingml.document");
        private static readonly MediaTypeHeaderValue Pptx = MediaTypeHeaderValue.Parse("application/vnd.openxmlformats-officedocument.presentationml.presentation");
        private static readonly MediaTypeHeaderValue Xlsx = MediaTypeHeaderValue.Parse("application/vnd.openxmlformats-officedocument.spreadsheetml.sheet");
        private static readonly MediaTypeHeaderValue Jpeg = MediaTypeHeaderValue.Parse("image/jpeg");
        private static readonly MediaTypeHeaderValue Png = MediaTypeHeaderValue.Parse("image/png");

        /* =========================================
         * Allow / Not Allow
         * ========================================= */

        public static bool IsNotAllowType(string mimeType) => !IsAllowType(mimeType);

        public static bool IsAllowType(string mimeType)
        {
            var type = MediaTypeHeaderValue.Parse(mimeType);

            return new[]
            {
                Pdf, Docx, Pptx, Xlsx, Jpeg, Png
            }.Any(allowed => Includes(allowed, type));
        }

        /* =========================================
         * Type Checks
         * ========================================= */

        public static bool IsPdf(string contentType) => Includes(Pdf, MediaTypeHeaderValue.Parse(contentType));

        public static bool IsDocx(string contentType) => Includes(Docx, MediaTypeHeaderValue.Parse(contentType));

        public static bool IsPptx(string contentType) => Includes(Pptx, MediaTypeHeaderValue.Parse(contentType));

        public static bool IsXlsx(string contentType) => Includes(Xlsx, MediaTypeHeaderValue.Parse(contentType));

        /* =========================================
         * MediaType includes 구현
         * ========================================= */

        private static bool Includes(MediaTypeHeaderValue parent, MediaTypeHeaderValue child)
        {
            return parent.Type.Equals(child.Type, StringComparison.OrdinalIgnoreCase)
                   && (parent.SubType == "*"
                       || parent.SubType.Equals(child.SubType, StringComparison.OrdinalIgnoreCase));
        }
    }
}